lib/elixir/test/elixir/kernel/warning_test.exs

```diff
@@ -28,6 +28,26 @@ defmodule Kernel.WarningTest do
     purge Sample
   end
 
+  test "unused variable in redefined function in different file" do
+    output = capture_err(fn ->
+      Code.eval_string """
+      defmodule Sample do
```

Comment:
This code does expose the bug because the warning would refer to ```no_file:3``` instead of ```supervisor.ex:5```. 
I like the assertion  ```assert output =~ "redefine_sample.ex:3"``` better because it seems more expressive and less brittle.